Description: Drop unneeded calls to CUPS server
 Don't try to contact the CUPS server when we can reliably
 determine that no printers are needed
Author: Steve Langasek <vorlon@debian.org>
Bug-Debian: http://bugs.debian.org/479512
Bug: https://bugzilla.samba.org/show_bug.cgi?id=5525
Forwarded: yes

Index: samba-3.6.0pre1/source3/smbd/server_reload.c
===================================================================
--- samba-3.6.0pre1.orig/source3/smbd/server_reload.c
+++ samba-3.6.0pre1/source3/smbd/server_reload.c
@@ -39,6 +39,10 @@
 	NTSTATUS status;
 	bool skip = false;
 
+	if (!lp_load_printers()
+	    && (lp_auto_services() == NULL || !strcmp(lp_auto_services(),"")))
+		return;
+
 	pcap_cache_reload();
 
 	status = make_server_info_system(talloc_tos(), &server_info);
